frontend/talentsearch/src/js/components/pool/PoolAdvertisementPage.tsx

```diff
@@ -770,6 +789,23 @@ const PoolAdvertisementPage = ({ id }: PoolAdvertisementPageProps) => {
     visible = true;
   }
 
+  // grab pool candidates of Me, then check whether a pool candidate exists that matches the advertisement AND is submitted
+  const meDataResponse = useGetMePoolCandidatesQuery();
+  const meData = meDataResponse[0]?.data;
```

Comment:
Access the data directly out of the array is not the best way to do this.  Check useGetPoolAdvertisementQuery on L781 above this for a better way.

Reply:
okay
here https://github.com/GCTC-NTGC/gc-digital-talent/pull/4319/commits/0e3ba54c346724b4e8488c729f8d15865567550b